GEOPM fails to build due to an unused variable in example code

**Describe the bug**
I tried to build GEOPM and the build failed due to compiler warnings about an unused variable in our example code.

**GEOPM version**
 0412dc996

**Expected behavior**
I expect the GEOPM build to succeed.

**Actual behavior**
The build failed due to an unused variable in an example file.

**Error messages produced**
```
examples/synthetic_benchmark.cpp:283:12: error: variable 'x' set but not used [-Werror,-Wunused-but-set-variable]
    double x = 0.0;
           ^
```

**Additional context**
I am using the OneAPI compiler (icpx), version 2022.0.0.

--- a/examples/synthetic_benchmark.cpp
+++ b/examples/synthetic_benchmark.cpp
@@ -335,7 +335,7 @@ void synthetic_benchmark_main(int nranks, int rank)
         geopm_tprof_post();
     }
     GET_TIME(tEnd);
-    fprintf(stderr, "%.2fs: Rank %d finished\n", tEnd-tStart, rank);
+    fprintf(stderr, "%.2fs: Rank %d finished. x = %f\n", tEnd-tStart, rank, x);
 
     geopm_time(&curr);
     geopm_prof_exit(region_id[0]);

